SonarQube.TeamBuild.PreProcessor/Roslyn/EmbeddedAnalyzerInstaller.cs

```diff
@@ -0,0 +1,208 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="EmbeddedAnalyzerInstaller.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using SonarQube.Common;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text;
+
+namespace SonarQube.TeamBuild.PreProcessor.Roslyn
+{
+    /// <summary>
+    /// Handles fetching embedded resources from SonarQube plugins
```

Comment:
It would be worth adding a note about the use of the caching mechanism (how / why)